validator/client/src/test/java/tech/pegasys/teku/validator/client/loader/ValidatorLoaderTest.java

```diff
@@ -282,6 +337,75 @@ void shouldEnableSlashingProtectionForLocalValidators(@TempDir Path tempDir) thr
             validator.getPublicKey(), forkInfo.getGenesisValidatorsRoot(), block.getSlot());
   }
 
+  @Test
+  void shouldLoadAdditionalExternalValidatorsOnReload() {
+    final PublicKeyLoader publicKeyLoader = mock(PublicKeyLoader.class);
+    final List<BLSPublicKey> initialKeys = List.of(PUBLIC_KEY1);
+    final String publicKeysUrl = "http://example.com";
+    when(publicKeyLoader.getPublicKeys(List.of(publicKeysUrl))).thenReturn(initialKeys);
+
+    final InteropConfig interopConfig = InteropConfig.builder().build();
+    final ValidatorConfig config =
+        ValidatorConfig.builder()
+            .validatorExternalSignerUrl(SIGNER_URL)
+            .validatorExternalSignerPublicKeySources(Collections.singletonList(publicKeysUrl))
+            .validatorExternalSignerSlashingProtectionEnabled(true)
+            .build();
+    final ValidatorLoader validatorLoader =
+        ValidatorLoader.create(
+            config,
+            interopConfig,
+            httpClientFactory,
+            slashingProtector,
+            publicKeyLoader,
+            asyncRunner,
+            metricsSystem);
+
+    validatorLoader.loadValidators();
+    final OwnedValidators validators = validatorLoader.getOwnedValidators();
+    assertThat(validators.getPublicKeys()).containsOnly(PUBLIC_KEY1);
+
+    final List<BLSPublicKey> reconfiguredKeys = List.of(PUBLIC_KEY1, PUBLIC_KEY2);
```

Comment:
Would it be worth adding some tests to see what happens when some keys are added + removed or even just removed?:
```suggestion
    final List<BLSPublicKey> reconfiguredKeys = List.of(PUBLIC_KEY2);
```

Reply:
Added a separate test that validators aren't removed so the requirement is clear.